voting/client-web/src/components/Header/Header.tsx

```diff
@@ -29,6 +30,20 @@ const HeaderImpl = ({ classes, isReadOnly }) => {
     { label: t('Rewards'), url: '/reward' },
   ];
 
+  function displayInMetamask() {
+    (window as any).web3.currentProvider.sendAsync({
```

Comment:
Maybe put all the web3 api calls in a service ?

Reply:
nice catch, fixed